Filtered_kernel/include/CGAL/Lazy.h

```diff
@@ -231,6 +229,11 @@ class Lazy_rep : public Rep, public Depth_base
 
   mutable AT at;
   mutable ET *et;
+#ifdef CGAL_LAZY_USE_MUTEX
+  typedef CGAL::cpp11::mutex Mutex;
+  typedef CGAL::cpp11::lock_guard<Mutex> Lock_guard;
+  mutable Mutex update_exact_mutex;
+#endif
```

Comment:
I agree the issue is how to update the approximate value.

Actually it does not have to be updated atomically: we need to update each of its two components (`inf` and `sup`), but if we update once atomically without updating the other, that is still a valid bounding interval.

On platforms were `std::atomic<double>` is lock-free (and they can be atomic on x86 and x86_64), we could implement a special version of `Internal_nt` were the two components are atomics.